### PR TITLE
Adding exception text to exit message for invalid clientsecrets

### PR DESCRIPTION
**Pull request by pferate**
This is to address a problem found in google/google-api-python-client#228.

This will give users the text from the InvalidClientSecretsError exception, in addition to the generic message that it currently does. The format of the message can be modified, if needed.

### CODE REVIEW

File: oauth2client/client.py [hunk 1]
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
                 client_info['client_id'], client_info['client_secret'],
                 scope, **constructor_kwargs)
 
-    except clientsecrets.InvalidClientSecretsError:
+    except clientsecrets.InvalidClientSecretsError as e:
         if message:
+            if str(e):
+                message = 'Error: %s\n================\n%s' % (str(e), message)
             sys.exit(message)
```
Comment (theacodes) [inline oauth2client/client.py:2217]: Holy cow, I'm really appalled we have a `sys.exit` hiding in here. D:

File: oauth2client/client.py [hunk 2]
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
                 client_info['client_id'], client_info['client_secret'],
                 scope, **constructor_kwargs)
 
-    except clientsecrets.InvalidClientSecretsError:
+    except clientsecrets.InvalidClientSecretsError as e:
         if message:
+            if str(e):
```
Comment (theacodes) [inline oauth2client/client.py]: `if e.args`?

File: oauth2client/client.py [hunk 3]
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
...
+                message = 'Error: %s\n================\n%s' % (str(e), message)
```
Comment (theacodes) [inline oauth2client/client.py]: I'd prefer just `'The client secrets were invalid: \n {}'.format(e)`

**Comment by jonparrott** [comment]
Thanks for this contribution. Do you mind adding or augmenting the existing test for this?

File: oauth2client/client.py (hunk 3 above)
Comment (pferate) [inline oauth2client/client.py]: Sounds better.  What about if message is passed as a parameter?

File: oauth2client/client.py (hunk 2 above)
Comment (pferate) [inline oauth2client/client.py]: better.  I'll update it.

Comment (theacodes) [inline oauth2client/client.py]: I'm unsure about py2.6 / py3 compatibility here so tread carefully.

File: oauth2client/client.py (hunk 3 above)
Comment (theacodes) [inline oauth2client/client.py]: Works just as well.

File: oauth2client/client.py (hunk 2 above)
Comment (pferate) [inline oauth2client/client.py]: Just tested this, it works in at least 2.6, 2.7, 3.2, 3.3, 3.4, and 3.5.

Comment (theacodes) [inline oauth2client/client.py]: Thanks.

File: oauth2client/client.py [hunk 4]
```diff
@@ -2210,8 +2210,10 @@ def flow_from_clientsecrets(filename, scope, redirect_uri=None,
                 client_info['client_id'], client_info['client_secret'],
                 scope, **constructor_kwargs)
 
-    except clientsecrets.InvalidClientSecretsError:
+    except clientsecrets.InvalidClientSecretsError as e:
         if message:
```
Comment (nathanielmanistaatgoogle) [inline oauth2client/client.py]: While we're here: `if message is not None:`.

Comment (pferate) [inline oauth2client/client.py]: will do.

**Comment by pferate** [comment]
@jonparrott @nathanielmanistaatgoogle, I updated this PR again and added a test for this (slightly modified a previous test).  Travis passed on my branch, but still waiting to launch for this PR.

I had to use positional argument specifiers in the `.format()` string to have it work with 2.6.

**Comment by jonparrott** [comment]
Great, this looks good to me. @nathanielmanistaatgoogle to do the final review.

**Comment by nathanielmanistaatgoogle** [comment]
Change content looks good; [please amend your commit log message to meet the guidelines](http://chris.beams.io/posts/git-commit/#seven-rules).

**Comment by pferate** [comment]
Commit message updated.

**Comment by jonparrott** [comment]
Waiting for travis.

**Comment by jonparrott** [comment]
Thanks for your contributions, @pferate!

**Pull request merged**